qiskit/visualization/pulse_v2/generators.py

```diff
@@ -0,0 +1,683 @@
+                                              meta=re_meta,
+                                              styles=re_style)
+        fill_objs.append(real)
+
+    # create imaginary part
+    if any(im_y):
+        im_style = style.copy()
+        im_style['color'] = color.imaginary
+        im_meta = meta.copy()
+        im_meta['data'] = 'imaginary'
+        imag = drawing_objects.FilledAreaData(data_type='Waveform',
+                                              channel=channel,
+                                              x=time,
+                                              y1=im_y,
+                                              y2=np.zeros_like(time),
+                                              meta=im_meta,
+                                              styles=im_style)
+        fill_objs.append(imag)
+
+    return fill_objs
+
+
+def gen_iqx_latex_waveform_name(inst_data: data_types.InstructionTuple) \
+        -> List[drawing_objects.TextData]:
+    r"""Generate formatted instruction name associated with the waveform.
+
+    Channel name and id are removed and the rotation angle is expressed in units of pi.
+    CR pulse name is also converted with the controlled rotation angle divided by 2.
+
+    Note that in many scientific articles the controlled rotation angle implies
+    the actual rotation angle, but in IQX backend the rotation angle represents
+    the difference between rotation angles with different control qubit states.
+
+    For example:
+        - 'X90p_d0_abcdefg' is converted into 'X(\frac{\pi}{2})'
+        - 'CR90p_u0_abcdefg` is converted into 'CR(\frac{\pi}{4})'
+
+    The `annotate` style is applied.
+
+    Notes:
+        This generator can convert pulse names used in the IQX backends.
+        If pulses are provided by the third party providers or the user defined,
+        the generator output may be the as-is pulse name.
+
+    Args:
+        inst_data: Waveform instruction data to draw.
+
+    Returns:
+        List of `TextData` drawing objects.
+    """
+    systematic_name = inst_data.inst.pulse.name
+
+    style = {'zorder': PULSE_STYLE.style['formatter.layer.annotate'],
+             'color': PULSE_STYLE.style['formatter.color.annotate'],
+             'size': PULSE_STYLE.style['formatter.text_size.annotate'],
+             'va': 'center',
+             'ha': 'center'}
+
+    template = r'(?P<op>[A-Z]+)(?P<angle>[0-9]+)?(?P<sign>[pm])_(?P<ch>[dum])[0-9]+'
+    match_result = re.match(template, systematic_name)
+    if match_result is not None:
+        match_dict = match_result.groupdict()
+        sign = '' if match_dict['sign'] == 'p' else '-'
+        if match_dict['op'] == 'CR':
+            # cross resonance
+            if match_dict['ch'] == 'u':
+                op_name = r'{\rm CR}'
+            else:
+                op_name = r'\overline{\rm CR}'
+            # IQX name def is not standard. Echo CR is annotated with pi/4 rather than pi/2
```

Comment:
Could you please elaborate on this for me?

Reply:
For example see our recent [paper about rotary tone](https://arxiv.org/pdf/2007.02925.pdf). In fig1(a) each pulse composing the echo sequence is represented by pi/4 which is 45 degree. However in IQX the pulse name becomes `CR90p_*`, i.e. when control angle is |0> (|1>) the rotation is +pi/4 (-pi/4) - I don't remember correct sign - so the difference of them is pi/2. This is convention of our pulse notation but very confusing. Perhaps we can change our pulse notation to de facto standard.